# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('../Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date']= pd.to_datetime(df['date'], infer_datetime_format=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [4]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])
df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [5]:
# Drop the date column
df = df.drop(columns='date')
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [50]:
# Filter data for June
june_data = df[df.index.month == 6]
june_data.head()

station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77

In [49]:
# Filter data for December
december_data = df[df.index.month == 12]
december_data.head()

station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64

In [51]:
# Identify the average temperature for June
june_data.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [52]:
# Identify the average temperature for December
december_data.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [53]:
# Create collections of temperature data
june_temp = june_data.iloc[::1,2]
june_temp

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [57]:
# Create collections of temperature data
december_temp = december_data.iloc[::1,2]
december_temp

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [55]:
# Run paired t-test
stats.ttest_ind(june_temp, december_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

* The difference in these two data sets is statistically significant because the p-value of 3.9e-191. is less than alpha (Typically < 0.05). 
*However, the T-statistic has a relatively small magnitude of 31.60 degrees farenheit meaning that the difference between the temperatures during June and December don't vary largely. 
In conclusion, Hawaii is a great place to vacation year round!

### Will you use a paired t-test, or an unpaired t-test? Why?
* I will use a paired t-test because I am comparing a group of data under two separate scenarios. In this case the temperature of Hawaii in different months of the year.